indico_zodbimport/modules/event_papers.py

```diff
@@ -169,6 +223,87 @@ def _migrate_paper_roles(self, contribution):
+        for old_answer in old_judgment._answers:
+            old_question = old_answer._question
+            question = self.legacy_question_map.get(old_question, None)
+            if not question:
+                self.importer.print_warning(cformat('%{yellow!}Answer to deleted question {} has been ignored [{}, {}]')
+                                            .format(old_question._id, contribution, review_type),
+                                            event_id=self.event.id)
+                continue
+
+            rating = PaperReviewRating(question=question, value=old_answer._rbValue)
+            review.ratings.append(rating)
+        return review
+
+    def _migrate_revisions(self, contribution, rm):
+        self.importer.print_info(cformat('%{white!}{}%{reset}').format(contribution.title))
+
+        self.file_checksums = defaultdict()
+
+        # Here we iterate over what the legacy PR mode calls "Reviews" (our `PaperRevisions`)
+        for n, old_revision in enumerate(rm._versioning, 1):
+            old_judgment = old_revision._refereeJudgement
+            old_content_reviews = old_revision._reviewerJudgements.values()
+            old_layout_review = old_revision._editorJudgement
+
+            # keep track of the last legacy revision, so that we can come back to it
+            # during paper file migration
+            self.legacy_contrib_last_revision_map[contribution.id] = old_revision
+
+            # skip revisions that haven't been submitted by the author
+            if not old_revision._isAuthorSubmitted:
+                # ... except if said revision has been judged before being marked as submitted (!)
+                if old_judgment._submitted:
+                    self.importer.print_warning(cformat('%{yellow!}Revision judged without being submitted! [{}: {}]')
+                                                .format(contribution, old_revision._version), event_id=self.event.id)
+                else:
+                    continue
+
+            # It seems contradictory, but 'submitted' in legacy means that there is a final decision
+            # We'll ignore legacy custom states and use TBC
+            state = (JUDGMENT_STATE_REVISION_MAP.get(int(old_judgment._judgement._id),
+                                                     PaperRevisionState.to_be_corrected)
+                     if old_judgment._submitted
+                     else PaperRevisionState.submitted)
+            judge = old_judgment._author.user if old_judgment._submitted else None
+            judgment_dt = _to_utc(old_judgment._submissionDate) if old_judgment._submitted else None
+            last_file = (LegacyPaperFile.query.with_parent(contribution)
+                         .filter(LegacyPaperFile.revision_id == old_revision._version)
+                         .order_by(LegacyPaperFile.created_dt.desc())
+                         .limit(1)).first()
```

Comment:
.limit(1)
    .first())

That's how we formatted it everythere else. Mostly a matter of preference though